packages/restapi/src/lib/spaceV2/acceptInvite.ts

```diff
@@ -0,0 +1,261 @@
+        const iceServerConfig = await getIceServerConfig(this.env);
+
+        let peerConnection = new Peer({
+            initiator: true,
+            trickle: false,
+            stream: this.data.local.stream,
+            config: {
+                iceServers: iceServerConfig,
+            },
+        });
+
+        this.setPeerConnection(recipientAddress, peerConnection);
+
+        peerConnection.on('error', (err: any) => {
+            if (this.data.incomingPeerStreams[0].retryCount >= 5) {
+                console.log('Max retries exceeded, please try again.');
+                this.disconnect({ peerAddress: recipientAddress });
+            }
+
+            // retrying in case of connection error
+            sendSpaceNotification(
+                {
+                    signer: this.signer,
+                    chainId: this.chainId,
+                    pgpPrivateKey: this.pgpPrivateKey,
+                },
+                {
+                    senderAddress,
+                    recipientAddress,
+                    status: VideoCallStatus.RETRY_INITIALIZED,
+                    spaceId: spaceId,
+                    signalData: null,
+                    env: this.env,
+                }
+            );
+        })
+
+        peerConnection.signal(signalData);
+
+        peerConnection.on('signal', (data: any) => {
+            this.setSpaceV2Data((oldData) => {
+                return produce(oldData, (draft) => {
+                    draft.meta.initiator.signal = data;
+                });
+            });
+
+            sendSpaceNotification(
+                {
+                    signer: this.signer,
+                    chainId: this.chainId,
+                    pgpPrivateKey: this.pgpPrivateKey,
+                },
+                {
+                    senderAddress,
+                    recipientAddress,
+                    status: retry
+                        ? VideoCallStatus.RETRY_RECEIVED
+                        : VideoCallStatus.INITIALIZED,
+                    spaceId,
+                    signalData: data,
+                    env: this.env,
+                    callDetails: details,
+                }
+            );
+        });
+
+        peerConnection.on('connect', () => {
+            peerConnection.send(
+                JSON.stringify({
+                    type: 'isVideoOn',
+                    value: this.data.local.video,
+                })
+            );
+            peerConnection.send(
+                JSON.stringify({
+                    type: 'isAudioOn',
+                    value: this.data.local.audio,
+                })
+            );
+
+            // set videoCallInfo state with status connected for the receiver's end
+            this.setSpaceV2Data((oldData) => {
+                return produce(oldData, (draft) => {
+                    const incomingIndex = getIncomingIndexFromAddress(
+                        oldData.incomingPeerStreams,
```

Comment:
this won't work, as we're swapping the peer from pending to incoming once stream event's callback is fired
and connect event is fired before stream, so ideally, here pendingPeerStreams will have that peer